R/eastafrica.R

```diff
@@ -0,0 +1,68 @@
+#' get KITE East Africa v2.1.csv from ...
+#'
+#' See:
+#' - A Database of Radiocarbon Dates for Palaeoenvironmental Research in Eastern Africa: https://www.openquaternary.com/articles/10.5334/oq.22/
+#' - Radiocarbon dates from eastern Africa in the CARD2.0 format: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/NJLNRJ
+#'
+#' The dataset is stored within https://dataverse.harvard.edu that requires the dataverse-package (see https://cran.r-project.org/web/packages/dataverse/vignettes/C-retrieval.html)
+#'
+#' @examples
+#'
+#' \dontrun{
+#'   KITEeastAfrica <- get_KITEeastAfrica()
+#' }
+#'
+#' @export
+get_KITEeastAfrica <- function() {
+
+  Sys.setenv("DATAVERSE_SERVER" = "dataverse.harvard.edu")
+
+  writeBin(get_file("CARD Upload Template - KITE East Africa v2.1.csv", "doi:10.7910/DVN/NJLNRJ"), "data-raw/KITE_EastAfrica.csv")
+
+  # read data
+  KITEeastafrica <- readr::read_csv("data-raw/KITE_EastAfrica.csv",
+      skip = 3,
+      trim_ws = TRUE,
+      col_types = list(
+        'Field Number'='_',
+        'Taxa Dated'= '_',
+        'Type of Date'= '_',
+        'Locality' = '_',
+        'Map Sheet' = '_',
+        'Elevation     (m ASL)' = '_',
+        'Submitter' = '_',
+        'Date Submitted' = '_',
+        'Collector' = '_',
+        'Date Collected' = '_',
+        'Updater' = '_',
+        'Date Updated' = '_',
+        'Measured Age' = '_',
+        'MA Sigma' = '_',
+        'Delta 13 C Source' = '_',
+        'Significance' = '_',
+        'Site Name' = '_',
+        'Stratigraphic Component' = '_',
+        'Context' = '_',
+        'Associated Taxa' = '_',
+        'Additional Information' = '_',
+        'Comments' = '_'
+        )) %>%
+    dplyr::rename(
+      labnr = .data[["Lab Number"]],
+      c14age = .data[["Normalized Age"]],
+      c14std = .data[["NA Sigma"]],
+      c13val = .data[["Delta 13C (per mil)"]],
+      material = .data[["Material Dated"]],
+      site = .data[["Site Identifier"]],
+      lat = .data[["Latitude"]],
+      lon = .data[["Longitude"]],
+      shortref = .data[["References"]]
```

Comment:
The additional information can also be part of the dataset. Just rename it accordingly (e.g. "Stratigraphic Component" -> stratigraphic_component or "Additional Information" ->comment (?)). 

We need a definite list of the variable names... An object of class c14_date_list can store any number of additional variables and is valid as long as the basic ones are present. And even here we have to be flexible.

Reply:
Concerning defined list of variable names, what do think about a simple markdown document that holds all variables currently pulled from all the different parsers? People could reference those and add new field if necessary? See https://github.com/dirkseidensticker/aDRAC/blob/master/data/aDRAC.md for what I've in mind.